ass5.cpp

```diff
@@ -0,0 +1,117 @@
+#include <iostream>
+#include <vector>
+#include <stack>
+
+using namespace std;
+
+#define ALPHABET_SIZE 50
+
+class Graph {
+    vector<vector<int> > adj_list;
+    
+public:
+    Graph () {
+        adj_list.resize(ALPHABET_SIZE);
+    }
+    
+    void addEdge(int v, int w) {
+        adj_list[v].push_back(w);
+    }
+    
+    void topologicalSortUtil(int v, vector<bool> &visited, stack<int> &stack);
+    
+    stack<int> topologicalSort(vector<int> letter_ids);
+};
+
+void Graph::topologicalSortUtil(int v, vector<bool> &visited, stack<int> &stack) {
+    visited[v] = true;
+    
+    for (auto i : adj_list[v]) {
+        if (!visited[i]) {
+            topologicalSortUtil(i, visited, stack);
+        }
+    }
+    
+    stack.push(v);
+}
+
+stack<int> Graph::topologicalSort(vector<int> letter_ids) {
+    stack<int> stack;
+    vector<bool> visited(letter_ids.size(), false);
+
+    for (auto i: letter_ids) {
+        if (!visited[i]) {
+            topologicalSortUtil(i, visited, stack);
+        }
+    }
+    
+    return stack;
+}
+
+string return_alphabet(vector<string> words) {
```

Comment:
So I guess if there is only one word and nothing else to compare with lexicographically, any letter order is possible. For now I am comparing it only with one variant, but probably it's better to make some assertion with all cases together.